app/controllers/curation_concerns/curation_concerns_controller.rb

```diff
@@ -33,6 +33,23 @@ def deny_access(exception)
     end
   end
 
+  def flag
```

Comment:
If the resource isn't flaggable, then the save will fail because of the state validation -- there's a test for this to make sure it won't let you flag a pending resource: https://github.com/pulibrary/plum/pull/283/files#diff-6259036f87e12d3642d5bcbcc3a36b15R386